fivetran_provider_async/hooks.py

```diff
@@ -134,6 +147,47 @@ async def get_sync_status_async(self, connector_id, previous_completed_at):
             job_status = "pending"
             return job_status
 
+    def pause_and_restart(self, connector_id, reschedule_for, reschedule_time):
+        """
+        While a connector is syncing, if it falls into a reschedule state,
+        wait for a time either specified by the user of recommended by Fivetran,
+        Then restart a sync
+
+        :param connector_id: Fivetran connector_id, found in connector settings
+            page in the Fivetran user interface.
+        :type connector_id: str
+        :param reschedule_for: From connector details, if schedule_type is manual,
+            then the connector expects triggering the event at the designated UTC time
+        :type reschedule_for: str
+        :param reschedule_time: Optional, if connector is in reset state
+            number of seconds to wait before restarting, else Fivetran suggestion used
+        :type reschedule_time: int
+        """
+        if reschedule_time:
+            log_statement = f'Starting connector again in "{reschedule_time}" seconds'
+            self.log.info(log_statement)
+            time.sleep(reschedule_time)
+        else:
+            wait_time = (
+                self._parse_timestamp(reschedule_for).add(minutes=1) - pendulum.now(tz="UTC")
+            ).seconds
+            log_statement = f'Starting connector again in "{wait_time}" seconds'
+            self.log.info(log_statement)
+            time.sleep(wait_time)
```

Comment:
Also when will this operator/method run? What if the rescheduled_for time has already elapsed and this operator is run after that. In that case, wait_time could be negative and this would raise an Exception, no?

Reply:
I think it would make sense if the rescheduled_for applies to manual sync too. As it looks tricky, if is it not a scheduled sync, then it should not also support reschedule, no?

Reply:
> Also when will this operator/method run? What if the rescheduled_for time has already elapsed and this operator is run after that. In that case, wait_time could be negative and this would raise an Exception, no?

I have done the change here. https://github.com/astronomer/airflow-provider-fivetran-async/pull/25/commits/202548d3de470c8ba2d3086c64cbb1082a27f6b7

Reply:
> I think it would make sense if the rescheduled_for applies to manual sync too as it looks tricky; if is it not a scheduled sync, then it should not also support rescheduling, no?

This is taken care here: https://github.com/astronomer/airflow-provider-fivetran-async/blob/32bd5d6ae37deb6c9816cbb38a7b90439df670df/fivetran_provider_async/hooks.py#L133